snail-kotlin/src/main/java/com/compass/snail/Observable.kt

```diff
@@ -147,12 +148,14 @@ open class Observable<T> : IObservable<T> {
         return observable
     }
 
-    override fun skip(first: UInt): Observable<T> {
+    override fun skip(first: Int): Observable<T> {
         val observable = Observable<T>()
-        var count = first.toInt()
+        var count = first
         subscribe(next = {
             if (count == 0) {
                 observable.next(it)
+            } else if (count < 0) {
+                observable.error(IllegalArgumentException("Parameter value must be nonzero"))
```

Comment:
Have we considered throwing a native exception as well? Or just defaulting to `0`?

Reply:
@bastienFalcou this is a native exception, I just added a friendly reason of why its invalid for the implementer

Reply:
@russellbstephens: I meant to literally `throw` it with the `throw` [keyword](https://kotlinlang.org/docs/reference/exceptions.html), just like an out of bound exception triggered when we try to access an invalid index in a list. 

Passing a negative value seems to be a development error and shouldn't happen, it doesn't seem crazy to raise an Exception in this case (although we might consider it too risky in case it would trigger crashes in edge cases we haven't encountered yet).